### PR TITLE
POA ignore region changes

### PR DESCRIPTION
**Pull request by chinmay-browserstack**
Implementation for POA.
+
Implementing Ignore Region for different types including ignore by id, xpath, selenium webelement and custom ignore region for POA.

### CODE REVIEW

File: Percy.Test/PercyDriver.Test.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+using Newtonsoft.Json;
+using RichardSzalay.MockHttp;
+using System.Net.Http;
+
+namespace PercyIO.Selenium.Tests
+{
+  public class PercyDriverTest
+  {
+    [Fact]
+    public void getDriverDetails()
```
Comment (Amit3200) [inline Percy.Test/PercyDriver.Test.cs:15]: change the function name for the test

File: Percy.Test/Percy.Test.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net5.0;net47</TargetFrameworks>
+    <TargetFrameworks>net7.0</TargetFrameworks>
```
Comment (samarsault) [inline Percy.Test/Percy.Test.csproj:4]: Why are we changing this?

Comment (chinmay-browserstack) [reply Percy.Test/Percy.Test.csproj:4]: This is needed as we are adding packages for unit test which need dotnet 7. also FYI appium SDK is also using 7 https://github.com/percy/percy-appium-dotnet/blob/main/Percy.Tests/Percy.Tests.csproj#L3

Comment (samarsault) [reply Percy.Test/Percy.Test.csproj:4]: Cool, .NET Target for Test project shouldn't affect SDK

File: Percy.Test/PercyDriver.Test.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      // Setup a respond for the user api
+      mockHttp.Expect(HttpMethod.Get, "http://localhost:5338/percy/healthcheck")
+        .Respond(header, "application/json", JsonConvert.SerializeObject(obj));
+      mockHttp.Expect(HttpMethod.Post, "http://localhost:5338/percy/automateScreenshot")
+        .WithPartialContent("Screenshot 1")
+        .Respond("application/json", JsonConvert.SerializeObject(obj));
+      Percy.setHttpClient(new HttpClient(mockHttp));
+      percyDriverMock.Object.Screenshot("Screenshot 1");
+
+      Mock<IWebElement> element = new Mock<IWebElement>();
```
[the remark this answers is not in the document]
Comment (samarsault) [reply Percy.Test/PercyDriver.Test.cs:62]: Yes +1

File: Percy/Percy.cs [hunk 4]
```diff
@@ -146,6 +165,62 @@ public class Options : Dictionary<string, object> {}
...
+                    {
+                        screenshotOptions.Add(o.Key, o.Value);
+                    }
+                }
+
+                if(options != null) {
+                    Dictionary<string, object> userOptions = options.ToDictionary(kv => kv.Key, kv => kv.Value);
+
+                    if(userOptions.ContainsKey(ignoreElementKey)) {
+                        var ignoreElements = userOptions[ignoreElementKey] as List<IWebElement>;
```
Comment (samarsault) [inline Percy/Percy.cs:201]: Lets not use `var` since we know the type `List<IWebElement>`

**Review by samarsault — APPROVED** [review]
LGTM

**Review by itsjwala — APPROVED** [review]

**Pull request merged**